include/effects/strip/fireeffect.h

```diff
@@ -412,6 +511,38 @@ class SmoothFireEffect : public LEDStripEffect
     {
     }
 
+    SmoothFireEffect(const JsonObjectConst& jsonObject)
+        : LEDStripEffect(jsonObject),
+          _Reversed(jsonObject[PTY_REVERSED]),
+          _Cooling(jsonObject[PTY_COOLING]),
+          _Sparks(jsonObject[PTY_SPARKS]),
+          _Drift(jsonObject["dft"]),
+          _DriftPasses(jsonObject["dtp"]),
+          _SparkHeight(jsonObject[PTY_SPARKHEIGHT]),
+          _Turbo(jsonObject["trb"]),
+          _Mirrored(jsonObject[PTY_MIRORRED])
+    {
+    }
+
+    virtual bool SerializeToJSON(JsonObject& jsonObject) 
+    {
+        StaticJsonDocument<128> jsonDoc;
+        
+        JsonObject root = jsonDoc.to<JsonObject>();
+        LEDStripEffect::SerializeToJSON(root);
+
+        jsonDoc[PTY_MIRORRED] = _Reversed;

```

Comment:
Why the mix of macro and const json property ids?  Are some well known and hence used across multiple effects, while the 'hardcoded' ones are one-offs?

Reply:
I agree that this is somewhat inconsistent. But yes, my consideration was exactly that.